infra/src/Pos/Infra/Communication/Types/Relay.hs

```diff
@@ -126,4 +134,44 @@ data RelayLogEvent =
     | EnqueueDequeueTime !Integer
     deriving Show
 
+instance Bi (DataMsg MCCommitment) where
+    encode (DataMsg (MCCommitment signedComm)) = encode signedComm
+    decode = DataMsg . MCCommitment <$> decode
+
+instance Bi (DataMsg MCOpening) where
+    encode (DataMsg (MCOpening sId opening)) = encodeListLen 2 <> encode sId <> encode opening
+    decode = do
+        enforceSize "DataMsg MCOpening" 2
+        DataMsg <$> (MCOpening <$> decode <*> decode)
+
+instance Bi (DataMsg MCShares) where
+    encode (DataMsg (MCShares sId innerMap)) = encodeListLen 2 <> encode sId <> encode innerMap
+    decode = do
+        enforceSize "DataMsg MCShares" 2
+        DataMsg <$> (MCShares <$> decode <*> decode)
+
+instance Bi (DataMsg MCVssCertificate) where
+    encode (DataMsg (MCVssCertificate vss)) = encode vss
+    decode = DataMsg . MCVssCertificate <$> decode
+
+-- TODO: someone who knows networking should take a look because this really
+-- doesn't look like something that anyone should ever have to write
```

Comment:
yeah, what on earth *is* this?

Reply:
I know!

That comment was there before I moved it. I left it in for now, because it documents that "there be dragons here". I actually suspect that after re-uniting orphans, that whole type can just be dropped.